src/igmpproxy.c

```diff
@@ -338,7 +338,7 @@ void igmpProxyRun(void) {
              * call gettimeofday.
              */
             if (Rt == 0) {
-                curtime.tv_sec = lasttime.tv_sec + secs;
+                curtime.tv_sec = lasttime.tv_sec + timeout->tv_sec;//timeout
```

Comment:
IIRC updating `timeout` value of pselect is Linux specific. So it would not work on other platforms correctly.

Reply:
Really manpage for pselect says:

On Linux, select() modifies timeout to reflect the amount of time not slept; **most other implementations do not do this**. (POSIX.1 permits either behavior.) This causes problems both when Linux code which reads timeout is ported to other operating systems, and when code is ported to Linux that reuses a struct timeval for multiple select()s in a loop without reinitializing it. **Consider timeout to be undefined after select() returns.**

So this patch is Linux-specific, but igmpproxy is used on more *BSD platforms.